test/fiddle/test_pointer.rb

```diff
@@ -20,16 +20,28 @@ def test_malloc_free_func_int
       assert_equal free.to_i, Fiddle::RUBY_FREE.to_i
 
       ptr  = Pointer.malloc(10, free.to_i)
-      assert_equal 10, ptr.size
-      assert_equal free.to_i, ptr.free.to_i
+      refute ptr.freed?
+      begin
+        assert_equal 10, ptr.size
+        assert_equal free.to_i, ptr.free.to_i
+      ensure
```

Comment:
Why do we need this?
The `ptr` is freed by GC when on of the above two assertions is failed,  isn't it?

Reply:
In my mind, it's good practice to free a native, external resource as soon as you can. That's why we use blocks for example when opening a file, rather than letting the GC close it. In the future I hope to create a PR for a similar block scope for pointers as well.

Reply:
Remove the manual free, leave it relying on GC for now, and then switch to the block malloc that's coming in the next PR? Ok.

Reply:
Done.

Reply:
Yes.